expression/builtin_string.go

```diff
@@ -1555,7 +1555,43 @@ type builtinQuoteSig struct {
 
 // See https://dev.mysql.com/doc/refman/5.7/en/string-functions.html#function_quote
 func (b *builtinQuoteSig) eval(row []types.Datum) (d types.Datum, err error) {
-	return d, errFunctionNotExists.GenByArgs("quote")
+	args, err := b.evalArgs(row)
+	if err != nil {
+		return types.Datum{}, errors.Trace(err)
+	}
+	if args[0].IsNull() {
+		return
+	}
+	var (
+		str    string
+		buffer bytes.Buffer
+	)
+	str, err = args[0].ToString()
+	if err != nil {
+		return d, errors.Trace(err)
+	}
+	runes := []rune(str)
+	buffer.WriteRune('\'')
+	for i := 0; i < len(runes); i++ {
```

Comment:
I guess this would work:
```
switch  runes[i] {
case 0, '\\', '\'':
    buffer.WriteRune('\\')
    buffer.WriteRune(runes[i])
case '\032':
    buffer.WriteRune('\\')
    buffer.WriteRune('Z')
default:
    buffer.WriteRune(runes[i]) 
}
```